packages/react-native-web/src/vendor/react-native/VirtualizedList/index.js

```diff
@@ -1317,7 +1317,9 @@ class VirtualizedList extends React.PureComponent<Props, State> {
   _onCellLayout(e, cellKey, index) {
     const layout = e.nativeEvent.layout;
     const next = {
-      offset: this._selectOffset(layout),
+      // OffsetTop gets the correct offset for each cell for inverted lists beacuse it ignores
+      // parent's transformations (like 'transform: [{scaleY: -1}]').
+      offset: !this.props.horizontal ? e.nativeEvent.target.offsetTop : layout.x,
```

Comment:
@necolas I have a question regarding your concern about this solution. If I want this to be merged do I have to find a solution that does the same but uses only the RN layout tools and not platform-specific API?

Another question: do you have an estimated date when the new react-native's VirtualizedList code will be applied to react-native-web?

Reply:
> If I want this to be merged do I have to find a solution that does the same but uses only the RN layout tools and not platform-specific API?

Ideally yes. But in this case it depends on whether the `y` value in RNW should be implemented differently to match RN. I still don't understand why `offsetTop` is used here but no where else. Is it because the layout values change on web with the scaleY transform, but don't in RN? Previously there have been suggestions that the scaleY approach breaks too many things on web anyway (like scroll wheel use).

This is why I want the canonical VirtualizedList implementation to have to support web, because it's too difficult to manage the necessary implementation across 2 projects when the upstream project doesn't think about web at all. Plus, VirtualizedList is slow on native and web, so RNW maintaining some fork of a slow thing (that most people shouldn't be using in its current state) is not a good use of time either.

> do you have an estimated date when the new react-native's VirtualizedList code will be applied to react-native-web?

No. But if this is blocking you, you could copy-paste it to RNW yourself and that updated will be part of the PR.

Reply:
> I still don't understand why offsetTop is used here but nowhere else. Is it because the layout values change on web with the scaleY transform, but don't in RN?

@necolas Yes, the `y` values of items change with [`transform: [{scaleY: -1}]`](https://github.com/necolas/react-native-web/blob/dee258ae02188befca4192ea502d4fbebc478341/packages/react-native-web/src/vendor/react-native/VirtualizedList/index.js#L2159) in Web. Meanwhile, in RN they don't change.

In RN, it doesn't matter if scaleY is normal or inverted, the first item will receive a `y` value equal to zero and the last item gets a `y` close to our list's length.

As I tried to explain in this PR's description, inside onLayout's Web implementation uses the DOM's [`getBoundingClientRect()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect). This API gets the positions of elements strictly relative to the viewport. Elements closer to the top of our screen will get `y` closer to zero and Elements further from the top will get a greater `y`.  When `transform: [{scaleY: -1}]` is applied, we position the first item of our list at the bottom, and Web's onLayout (with`getBoundingClientRect()`) returns the greatest `y` value for our first item. The last item, at the top of our VirtualizedList, will receive a `y` equal to zero. But as more elements are rendered, the new elements at the top of our inverted list will get a `y` value also equal to zero. That will make a mess in the calculation of virtualization.

`offsetTop` gets the `y` relative to the parent. It ignores the parent's transformations ( like `transform: [{scaleY: -1}]`). But in order to work the parent must be a positioned element i.e. `position: relative` or `position: absolute`.

This is an edge case where Web's onLayout is not working as its native counterpart. Basically, `offsetTop` here is patching a Web's onLayout edge case. An edge case caused by its functionality being affected by element transformations.

Reply:
I agree this change in particular seems like a platform difference, but I think you need to account for horizontal inverted lists as well?

```suggestion
      offset: !horizontalOrDefault(this.props.horizontal) ? e.nativeEvent.target.offsetTop : e.nativeEvent.target.offsetLeft,
```